packages/client/components/TeamPrompt/WorkDrawer/GCalIntegrationPanel.tsx

```diff
@@ -0,0 +1,117 @@
+import React, {useState} from 'react'
+import graphql from 'babel-plugin-relay/macro'
+import {useFragment} from 'react-relay'
+import {GCalIntegrationPanel_meeting$key} from '../../../__generated__/GCalIntegrationPanel_meeting.graphql'
+import useAtmosphere from '../../../hooks/useAtmosphere'
+import useMutationProps from '../../../hooks/useMutationProps'
+import GCalIntegrationResultsRoot from './GCalIntegrationResultsRoot'
+import GcalClientManager from '../../../utils/GcalClientManager'
+import gcalSVG from '../../../styles/theme/images/graphics/google-calendar.svg'
+import clsx from 'clsx'
+
+const GCAL_QUERY_TABS = [
+  {
+    key: 'past7d',
+    label: 'Past 7 days'
+  },
+  {
+    key: 'today',
+    label: 'Today'
+  },
+  {
+    key: 'upcoming',
+    label: 'Upcoming'
+  }
+] as const
+
+interface Props {
+  meetingRef: GCalIntegrationPanel_meeting$key
+}
+
+const GCalPanel = (props: Props) => {
+  const {meetingRef} = props
+  const meeting = useFragment(
+    graphql`
+      fragment GCalIntegrationPanel_meeting on TeamPromptMeeting {
+        viewerMeetingMember {
+          teamMember {
+            teamId
+            integrations {
+              gcal {
+                auth {
+                  providerId
+                }
+                cloudProvider {
+                  id
+                  clientId
+                }
+              }
+            }
+          }
+        }
+      }
+    `,
+    meetingRef
+  )
+
+  const teamMember = meeting.viewerMeetingMember?.teamMember
+
+  const [eventRangeKey, setEventRangeKey] = useState<'past7d' | 'today' | 'upcoming'>('past7d')
+
+  const atmosphere = useAtmosphere()
+  const mutationProps = useMutationProps()
+
+  const gcal = teamMember?.integrations.gcal
+
+  const authGCal = () => {
+    if (!gcal?.cloudProvider || !teamMember) {
+      return
+    }
```

Comment:
+1 as long as I understand it _should_ not happen, it happened to me while reviewing the code as I didn't have the gcal env variables set. console warning here would save me some time :)

Reply:
Good point! Added some errors like we do on the GH + Jira panels